tests/testthat/test-profile_emissions.R

```diff
@@ -456,7 +386,7 @@ test_that("with some match preserves unmatched products (#193)", {
   expect_equal(unique(product$activity_uuid_product_uuid), c("unmatched", uuid))
 })
 
-test_that("with no match preserves unmatched products (#193)", {
+test_that("with no match errors gracefully", {
```

Comment:
Here I completely change the goal of this test. The old one tested that unmatched products are preserved when the data contained only unmatched products. That type of data is no longer valid and instead we throw an error. We need at least one matched product for the data to be valid. 

We still have a test to ensure unmatched products are preserved ([here](https://github.com/2DegreesInvesting/tiltIndicatorAfter/blob/49206c80d2287ba8ed1e735633496c88a59632aa/tests/testthat/test-profile_emissions.R#L421)), but the test data must have at least one matched product.

Reply:
In this contex "test data" means the data used in this specific test.